### ISSUE TITLE
Enabling of AppArmor feature by default in Security Profile Operator

### ISSUE THREAD

**Issue by juggarnautss** [kubernetes-sigs/security-profiles-operator#1320]
Requirement : 
Enabling AppArmor feature by default in Security Profile Operator installation.

Current behavior is that we have to first install the Security Profile Operator , after then we have to patch the spod with appararmor flag to turn on the apparmor feature.

In our project, we will be always using the AppArmor feature by default whenever we are installing the Security Profile Operator on K8S.
In short, the requirement is when SPO is installed apparmor is enabled whereas seccomp , selinux will be disabled by default.
Hence this feature request.

cc: @pjbgf

**Comment by pjbgf** [comment]
I propose we expand our helm chart to expose values to enable users to manage the `SecurityProfilesOperatorDaemon` config, so that users can toggle what's deployed more easily via three new values: `enableSelinux`, `enableLogEnricher` and `enableAppArmor`.

In terms of deploying components by default, both SELinux and AppArmor have underlying requirements at the worker nodes, so I think it would make sense to be an explicit decision from users when to enable each one of them. Arguably, even the log enrichment feature would just keep failing if the path to the audit logs is not correctly set.

Overall, I think users would benefit by having as straightforward deployment experience as possible. But at the same time, I would rather default to not running components, than to have the possibility of users getting errors (or crashing components) as their first experience with the SPO.

Any thoughts on this @JAORMX @saschagrunert @jhrozek?

**Comment by saschagrunert** [comment]
@pjbgf I agree! We also cannot automate it away since worker nodes would come with different configurations. We could still improve by pre-checking which features are supported on which nodes and then report the result somehow to the end users. Maybe this is worth another investigation.

**Comment by jhrozek** [comment]
On Sun, Nov 27, 2022 at 12:08:01AM -0800, Paulo Gomes wrote:
> Any thoughts on this @JAORMX @saschagrunert @jhrozek?

I agree. In terms of enabling a feature at deployment time (which from
the user's POV would be the same as feature enabled by default), at
least OLM allows you to set an environment variable during installation,
so if the feature can be enabled via an env var, then that's all that's
needed.

**Comment by rahulroshank** [comment]
@pjbgf We are on a verge of packaging SPO in our project. Wanted to know an approx. schedule for this issue closure.

**Comment by rahulroshan-kachchap** [comment]
Hi @pjbgf ,
Can you point out the location (which file) under helm charts to add the enableAppArmor: true parameter ?
Also, let us know how spod and security-profiles-operator pods are getting deployed as i didnt find their instances under template deployment.
I have tried adding default: true in crd.yml under 
enableAppArmor:
                description: tells the operator whether or not to enable AppArmor
                  support for this SPOD instance.
                type: boolean
                default: true
this is not enabling the apparmor during the helm chart install.

**Comment by pjbgf** [comment]
@rahulroshan-kachchap nice one, thank you for working on this. I added some comments on the PR itself, but basically once the commits are squashed we should be good to merge it. :+1:

**Comment by rahulroshan-kachchap** [comment]
Thanks @pjbgf. I have few question:
- In our solution, can we control the mode (enforce/complain) for the apparmor profiles we are assigning to containers ?
- Assuming it to be No, Can i go ahead and create an issue for the same.

Solution to this as per my understanding will be:
- need to add extra required property value in the CRD of AppArmorProfile like below:
  properties:
            enforced:
              type: boolean
              description: Determine whether the AppArmor profile is enforced
- and then adding the logic in the apparmorprofile.go code to pick this attribute value as assigned by the user and use during the applying/loading of profile.
Waiting for your views on this.

**Comment by pjbgf** [comment]
Added my comments on the feature request issue: https://github.com/kubernetes-sigs/security-profiles-operator/issues/1399.
